  INFO root [2019-05-18 14:48:28,203] - 
  INFO root [2019-05-18 14:48:28,204] - -----------------------------------------------------------------------------------------


  INFO root [2019-05-18 14:48:28,205] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-05-18 14:48:28,205] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/HappyPoker_assets/;D:/kbengine/HappyPoker_assets/scripts/;D:/kbengine/HappyPoker_assets/res/
  INFO root [2019-05-18 14:48:28,205] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-05-18 14:48:28,216] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-05-18 14:48:28,217] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-05-18 14:48:28,220] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:59250 )
  INFO root [2019-05-18 14:48:28,220] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:59250, SOMAXCONN=128.
  INFO root [2019-05-18 14:48:28,221] - server-configs:
  INFO root [2019-05-18 14:48:28,221] - 	internalTcpAddr : 192.168.56.1:59250
  INFO root [2019-05-18 14:48:28,221] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-05-18 14:48:28,221] - 	componentID : 9000
  WARN root [2019-05-18 14:48:28,380] - invalid UID(0) <= 0, please check UID for environment! automatically set to 3960.
  INFO root [2019-05-18 14:48:28,454] - Components::findLogger: found logger, addr:192.168.56.1:59221
  INFO root [2019-05-18 14:48:28,455] - Components::addComponent: reset orderLog, uid=3960!
  INFO root [2019-05-18 14:48:28,455] - Components::addComponent[logger], uid=3960, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-05-18 14:48:28,463] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 3960. PID: 5448 ----
  INFO root [2019-05-18 14:48:28,464] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-05-18 14:48:28,464] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-05-18 14:48:28,465] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-05-18 14:48:28,668] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-05-18 14:48:28,668] - Script::install(): KBExtra is successfully!
S_INFO root [2019-05-18 14:48:28,719] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-05-18 14:48:28,720] - TelnetServer server is running on port 31000
  INFO root [2019-05-18 14:48:29,084] - ---- loginapp is running ----
 DEBUG root [2019-05-18 14:48:29,085] - Components::process(): Request for the process of identity...
  WARN root [2019-05-18 14:48:35,182] - Forwarding logs to logger[192.168.56.1:59221/0/0/0]...
 ERROR root [2019-05-18 14:51:36,089] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=170955128061, lastReceivedTime=170916415624, inactivityExceptionPeriod=134887676)
  INFO root [2019-05-18 14:51:36,091] - The following logs sent to logger failed:
  WARN root [2019-05-18 14:51:36,091] - ==> [2019-05-18 14:51:36 090] TCPPacketReceiver::processPendingEvents(192.168.56.1:59221): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-05-18 14:51:36,091] - ==> [2019-05-18 14:51:36 090] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=170955128061, lastReceivedTime=170916415624, inactivityExceptionPeriod=134887676)
  WARN root [2019-05-18 14:51:36,221] - TCPPacketReceiver::processPendingEvents(192.168.56.1:59224): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-05-18 14:51:36,224] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=170955430850, lastReceivedTime=170892829201, inactivityExceptionPeriod=134887676)
  INFO root [2019-05-18 14:51:36,226] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-05-18 14:51:36,227] - onLoginAppShutDown()
  INFO root [2019-05-18 14:51:36,227] - Shutdowner::onShutdown: shutting down
  INFO root [2019-05-18 14:51:36,228] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-05-18 14:51:36,235] - Script::uninstall(): is successfully!
 DEBUG root [2019-05-18 14:51:36,235] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-05-18 14:51:36,536] - ThreadPool::destroy(): waiting for thread(2)[0x214d958,0x214d6e8,], try=1
